tools/regression_test.py

```diff
@@ -0,0 +1,674 @@
+        deploy_config_dir (str): Deploy config directory.
+        checkpoint_path (Path): Checkpoints path.
+        work_dir (Path): A working directory.
+        device (str): A string specifying device, defaults to '0'.
+        pytorch_metric (dict): All pytorch metric info.
+        metric_tolerance (dict):Tolerance for metrics.
+        report_dict (dict): Report info dict.
+        logger (logging.Logger): Logger.
+        backend_name (str):  Backend name.
+
+    Returns:
+        Dict: metric info of the model
+    """
+
+    backends_info = backends_info.get(backend_name, [])
+    if len(backends_info) <= 0:
+        return {}
+
+    backend_file_info = {
+        'onnxruntime': 'end2end.onnx',
+        'tensorrt': 'end2end.engine',
+    }
+
+    metric_info_dict = {
+        # mmdet
+        'bbox': {
+            'meta_name': 'box AP',
+            'metric_name': 'bbox_mAP',
+        },
+        'segm': {
+            'meta_name': 'mask AP',
+            'metric_name': '?',
+        },
+        'proposal': {
+            'meta_name': 'PQ',
+            'metric_name': '?',
+        },
+    }
+
+    performance_align = backends_info.get('performance_align', False)
+
+    metric_name_list = [str(metric) for metric in pytorch_metric]
+    assert len(metric_name_list) > 0
+
+    metric_all_list = [str(metric) for metric in metric_tolerance]
+    deploy_cfg_path_list = backends_info.get('deploy_config')
+
+    for infer_type, deploy_cfg_info in deploy_cfg_path_list.items():
+        for fp_size, deploy_cfg_name in deploy_cfg_info.items():
+            if deploy_cfg_name is None:
+                continue
+
+            metric_list = []
+
+            device_type = 'cuda' if device != '-1' else 'cpu'
+            deploy_cfg_path = Path(deploy_config_dir,
+                                   deploy_cfg_name).absolute()
+
+            backend_output_path = Path(work_dir). \
+                joinpath(Path(checkpoint_path).parent.parent.name,
+                         Path(checkpoint_path).parent.name,
+                         backend_name,
+                         infer_type,
+                         Path(checkpoint_path).stem)
+            backend_output_path = backend_output_path.absolute().resolve()
+            backend_output_path.mkdir(parents=True, exist_ok=True)
+
+            cmd_str = f'cd {str(Path().cwd())} && ' \
+                      'python3 ./tools/deploy.py ' \
+                      f'{str(deploy_cfg_path.absolute().resolve())} ' \
+                      f'{str(model_cfg_path.absolute().resolve())} ' \
+                      f'{str(checkpoint_path.absolute().resolve())} ' \
+                      './tests/data/tiger.jpeg ' \
+                      f'--work-dir {backend_output_path} ' \
+                      f'--device {device_type} ' \
+                      '--log-level INFO'
+
+            print(f'Process cmd = {cmd_str}')
+
+            # Convert the model to specific backend
+            os.system(cmd_str)
```

Comment:
what if running cmd failed, should we try-catch here?

Reply:
Done